### PR TITLE
NEW (Extension) @W-15639759@ Introduce a new command to run graph engine at the project level

### PR DESCRIPTION
**Pull request by jag-j**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -162,6 +166,10 @@
                     "command": "sfca.runDfaOnSelectedMethod",
                     "when": "false"
                 },
+                {
+                    "command": "sfca.runDfa",
+                    "when": "false"
```
Comment (jag-j) [inline package.json:171]: I'm setting this as false so the new command will not show up until Delta run is fully complete. To test this locally, you'll have to change this to true.

File: src/extension.ts [hunk 2]
```diff
@@ -120,19 +120,89 @@ export async function activate(context: vscode.ExtensionContext): Promise<vscode
...
+		});
+		customCancellationToken = new vscode.CancellationTokenSource();
+		customCancellationToken.token.onCancellationRequested(async () => {
+			customCancellationToken?.dispose();
+			customCancellationToken = null;
+			await vscode.window.showInformationMessage(messages.graphEngine.noViolationsFound);
+			return;
+		});
+
+		const projectDir: string[] = vscode.workspace.workspaceFolders?.map(folder => folder.uri.path);
```
Comment (jag-j) [inline src/extension.ts:186]: workspaceRoot is now deprecated and this seems to be the way to get the project directory's root. I couldn't get much info on this from the docs, but I have tested this and it works as expected.

File: src/extension.ts [hunk 3]
```diff
@@ -120,19 +120,89 @@ export async function activate(context: vscode.ExtensionContext): Promise<vscode
...
+		await _runDfa(context, outputChannel);
+	});
+	context.subscriptions.push(runOnActiveFile, runOnSelected, runDfaOnSelectedMethodCmd, runDfaOnWorkspaceCmd, removeDiagnosticsOnActiveFile, removeDiagnosticsOnSelectedFile, removeDiagnosticsInRange);
 	TelemetryService.sendExtensionActivationEvent(extensionHrStart);
 	outputChannel.appendLine(`Extension sfdx-code-analyzer-vscode activated.`);
 	return Promise.resolve(context);
 }
 
+async function _runDfa(context: vscode.ExtensionContext, outputChannel: vscode.LogOutputChannel) {
+	const choice = await vscode.window.showQuickPick(
```
Comment (stephen-carter-at-sf) [inline src/extension.ts:146]: Does this prompt show up every time - even when you haven't ran before? Ideally it should only show up when we have ran already with violations.

File: src/extension.ts [hunk 4]
```diff
@@ -120,19 +120,89 @@ export async function activate(context: vscode.ExtensionContext): Promise<vscode
...
+		customCancellationToken.token.onCancellationRequested(async () => {
+			customCancellationToken?.dispose();
+			customCancellationToken = null;
+			await vscode.window.showInformationMessage(messages.graphEngine.noViolationsFound);
+			return;
+		});
+
+		const projectDir: string[] = vscode.workspace.workspaceFolders?.map(folder => folder.uri.path);
+		if (projectDir.length === 0) {
+			void vscode.window.showWarningMessage('***No project directory could be identified. Not proceeding with DFA run.***');
```
Comment (stephen-carter-at-sf) [inline src/extension.ts:188]: When is this possible? Is it when someone opens up vscode for a file and then closes the file or something?

File: src/extension.ts (hunk 2 above)
Comment (stephen-carter-at-sf) [reply src/extension.ts:186]: Should we move this logic inside of targeting.getProjectDir() and make currentFile an optional input argument?

Comment (jag-j) [reply src/extension.ts]: That's good idea. Done.

File: src/extension.ts (hunk 3 above)
Comment (jag-j) [reply src/extension.ts]: As discussed in the standup, took care of this to show only when cache exists.

File: src/extension.ts (hunk 4 above)
Comment (jag-j) [reply src/extension.ts]: This happens when you haven't opened up any project and still try to run SFCA commands. This is something we should fix at a later point. For example, some of the other extensions make sure a sfdx project is loaded before showing their commands.

**Review by jfeingold35 — APPROVED** [review]

**Pull request merged**